src/components/time.rs

```diff
@@ -46,6 +47,69 @@ impl Time {
+        largest_unit: Option<TemporalUnit>,
+        smallest_unit: Option<TemporalUnit>,
+    ) -> TemporalResult<Duration> {
+        // 1. If operation is SINCE, let sign be -1. Otherwise, let sign be 1.
+        // 2. Set other to ? ToTemporalTime(other).
+        // 3. Let resolvedOptions be ? SnapshotOwnProperties(? GetOptionsObject(options), null).
+        // 4. Let settings be ? GetDifferenceSettings(operation, resolvedOptions, TIME, « », "nanosecond", "hour").
+        let increment = utils::to_rounding_increment(rounding_increment)?;
+        let (sign, rounding_mode) = if op {
+            (
+                -1.0,
+                rounding_mode
+                    .unwrap_or(TemporalRoundingMode::Trunc)
+                    .negate(),
+            )
+        } else {
+            (1.0, rounding_mode.unwrap_or(TemporalRoundingMode::Trunc))
+        };
+
+        let smallest_unit = smallest_unit.unwrap_or(TemporalUnit::Nanosecond);
+        // Use the defaultlargestunit which is max smallestlargestdefault and smallestunit
+        let largest_unit = largest_unit.unwrap_or(smallest_unit.max(TemporalUnit::Hour));
+
+        // 5. Let norm be ! DifferenceTime(temporalTime.[[ISOHour]], temporalTime.[[ISOMinute]],
+        // temporalTime.[[ISOSecond]], temporalTime.[[ISOMillisecond]], temporalTime.[[ISOMicrosecond]],
+        // temporalTime.[[ISONanosecond]], other.[[ISOHour]], other.[[ISOMinute]], other.[[ISOSecond]],
+        // other.[[ISOMillisecond]], other.[[ISOMicrosecond]], other.[[ISONanosecond]]).
+        let time = self.iso.diff(&other.iso);
+
+        // 6. If settings.[[SmallestUnit]] is not "nanosecond" or settings.[[RoundingIncrement]] ≠ 1, then
+        let norm = if smallest_unit != TemporalUnit::Nanosecond || rounding_increment != Some(1.0) {
+            // a. Let roundRecord be ! RoundDuration(0, 0, 0, 0, norm, settings.[[RoundingIncrement]], settings.[[SmallestUnit]], settings.[[RoundingMode]]).
+            let round_record = time.round(increment, smallest_unit, rounding_mode)?;
+            // b. Set norm to roundRecord.[[NormalizedDuration]].[[NormalizedTime]].
+            round_record.0
+        } else {
+            time.to_normalized()
+        };
+
+        // 7. Let result be BalanceTimeDuration(norm, settings.[[LargestUnit]]).
+        let result = TimeDuration::from_normalized(norm, largest_unit)?.1;
+        // 8. Return ! CreateTemporalDuration(0, 0, 0, 0, sign × result.[[Hours]], sign × result.[[Minutes]], sign × result.[[Seconds]], sign × result.[[Milliseconds]], sign × result.[[Microseconds]], sign × result.[[Nanoseconds]]).
+        Duration::new(
+            0.0,
+            0.0,
+            0.0,
+            0.0,
+            sign * result.hours,
+            sign * result.minutes,
+            sign * result.seconds,
+            sign * result.milliseconds,
+            sign * result.microseconds,
+            sign * result.nanoseconds,
+        )
+    }
```

Comment:
I think in this case it would be better to extract the sign and rounding type calculation outside of the function, and just do that on the specific `until` and `since` methods. It would make this a lot cleaner.

Reply:
Yeah, I was actually going to post that as a potential issue for discussion soon regarding handling those options. I think that entire section needs to be probably discussed as to how to handle it.

Right now the options parameters are separate and there is a `RelativeTo` parameter on `round` functions. I wasn't the biggest fan of that structure, and after playing around with it and review the recent PRs involving it. I think it would be better to remove `RelativeTo`, but at the same time, to still simplify the function parameters, add a `DifferenceSettings` struct. All the lines from 65-79 could then be extracted into a method on `DifferenceSettings` something like `difference_settings.prepare()`.

But this is mostly initial ideation on my part. Any thoughts?

Reply:
That would be pretty cool! Would also allow extending the function with more options in the future without a breaking change, which is really nice :)

Reply:
Since that change is on the plans then, I think you can ignore this review and do the changes on a separate PR.